salt/states/saltmod.py

```diff
@@ -689,6 +732,137 @@ def runner(name, **kwargs):
     return ret
 
 
+def parallel_runners(name, runners):
+    '''
+    Executes multiple runner modules on the master in parallel.
+
+    .. versionadded:: 2017.x.0 (Nitrogen)
+
+    A separate thread is spawned for each runner. This state is intended to be
+    used with the orchestrate runner in place of the ``saltmod.runner`` state
+    when different tasks should be run in parallel. In general, Salt states are
+    not safe when used concurrently, so ensure that they are used in a safe way
+    (e.g. by only targeting separate minions in parallel tasks).
+
+    name:
+        name identifying this state. The name is provided as part of the
+        output, but not used for anything else.
+
+    runners:
+        list of runners that should be run in parallel. Each element of the
+        list has to be a dictionary. This dictionary's name entry stores the
+        name of the runner function that shall be invoked. The optional kwarg
+        entry stores a dictionary of named arguments that are passed to the
+        runner function.
+
+    .. code-block:: yaml
+
+        parallel-state:
+           saltext.parallel-runner:
+             - runners:
+               - name: state.orchestrate
+                 kwarg:
+                   mods: orchestrate_state_1
+               - name: state.orcestrate
+                 kwarg:
+                   mods: orchestrate_state_2
+    '''
+    try:
+        jid = __orchestration_jid__
+    except NameError:
+        log.debug(
+            'Unable to fire args event due to missing __orchestration_jid__')
+        jid = None
+
+    def call_runner(runner_config):
+        return __salt__['saltutil.runner'](runner_config['name'],
+                                           __orchestration_jid__=jid,
+                                           __env__=__env__,
+                                           full_return=True,
+                                           **(runner_config['kwarg']))
```

Comment:
Should account for runners that don't require args. Perhaps `runner_config.get('kwarg', {})`.